pkg/scheduler/scheduler.go

```diff
@@ -367,6 +367,20 @@ func (sched *Scheduler) bind(assumed *v1.Pod, b *v1.Binding) error {
 	// If binding succeeded then PodScheduled condition will be updated in apiserver so that
 	// it's atomic with setting host.
 	err := sched.config.GetBinder(assumed).Bind(b)
+	if err != nil {
```

Comment:
No need to add this block. The error handling starts at line 373 of the original code. Whether an error occurs or not, we always call `sched.config.SchedulerCache.FinishBinding` and we start error handling after the call.

Reply:
I didn't say that we should ignore the returned error. I said that the error is checked and processed at line 373. So, we already have the logic to process the error.

Reply:
After @bsalamat's comment, I took a close look at the original code. Indeed, the `err` declared on line 370 is scoped to that if-block. So this err *is* checked, but the code is easy to mis-read.

I am trying to explain the current logic to myself, and I am just confused. I would welcome a PR (this PR?) that makes this logic easier to read.

Reply:
No, we shouldn't. If we do and "ForgetPod" fails, the pod will get stuck in the cache forever. I think the best course of action at this point is to rename the `err` returned by `FinishBinding` to `finErr` or something similar to avoid the confusion and leave the rest as is.

Reply:
sounds good, i will make that small fix for now and take a deeper look at how this GetBinder , FinishBinding is designed to need such a pattern